### ISSUE TITLE
Question: Is it possible to manage iOS & android caps in separate  serenity.properties files

### ISSUE THREAD

**Issue by vikramvi** [serenity-bdd/serenity-core#1163]
- Current setup has got single serenity.properties file with 2 set of values for iOS and Android
- Before run I need to comment either of these 2 set of caps manually
- Is there any way to keep both platform caps and switch to tell which one can be used for particular test run ?  
- Or is it possible to maintain 2 separate serenity.properties so that in saucelabs both test cases can be run in parallel ?

```
####### iOS CAPS ######
#appium.automationName = XCUITest
#appium.platformName=IOS
#appium.platformVersion = 11.0
#appium.deviceName  = iPhone SE
#appium.app = /Users/vikramanna/Documents/vikram/serenity-core/serenity-appium/src/test/java/integration/resources/WordPress.app
#appium.app = sauce-storage:WordPress.zip

####### Android CAPS ######
appium.automationName=uiautomator2
appium.platformName = Android
appium.platformVersion = 6.0
appium.deviceName  = Android Emulator
#appium.app =  /Users/vikramanna/Documents/vikram/serenity-core/serenity-appium/src/test/java/integration/resources/WordPress-debug.apk
appium.app =  sauce-storage:WordPress-debug.apk
appium.appPackage = org.wordpress.android
appium.appActivity = org.wordpress.android.ui.accounts.LoginActivity

```

**Comment by wakaleo** [comment]
If you only use one set of properties for any given test run, you could use Maven profiles for this.

**Comment by vikramvi** [comment]
Thanks @wakaleo do you have any example about Maven profiles ? With maven profile can I keep both set of properties, it seems interesting solution to implement. Thanks in advance.

**Comment by wakaleo** [comment]
See http://maven.apache.org/guides/introduction/introduction-to-profiles.html

**Comment by vikramvi** [comment]
Thanks, will read it and try to fix it before submitting new PR

**Comment by vikramvi** [comment]
@wakaleo I have tried to use profile as below

```
<?xml version="1.0" encoding="UTF-8"?>
<project xmlns="http://maven.apache.org/POM/4.0.0" xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
         xsi:schemaLocation="http://maven.apache.org/POM/4.0.0 http://maven.apache.org/xsd/maven-4.0.0.xsd">
    <modelVersion>4.0.0</modelVersion>

    <groupId>net.serenity-bdd.appium.integration</groupId>
    <artifactId>serenity-appium</artifactId>
    <version>1.0-SNAPSHOT</version>
    <packaging>jar</packaging>

    <name>Serenity Appium Integration Test Cases</name>

    <properties>
        <project.build.sourceEncoding>UTF-8</project.build.sourceEncoding>
        <serenity.version>1.9.3</serenity.version>
        <serenity.cucumber.version>1.9.3</serenity.cucumber.version>
        <serenity.maven.version>1.8.21</serenity.maven.version>
    </properties>

    <repositories>
        <repository>
            <snapshots>
                <enabled>false</enabled>
            </snapshots>
            <id>central</id>
            <name>bintray</name>
            <url>http://jcenter.bintray.com</url>
        </repository>
    </repositories>
    <pluginRepositories>
        <pluginRepository>
            <snapshots>
                <enabled>false</enabled>
            </snapshots>
            <id>central</id>
            <name>bintray-plugins</name>
            <url>http://jcenter.bintray.com</url>
        </pluginRepository>
    </pluginRepositories>

    <dependencies>
        <dependency>
            <groupId>net.serenity-bdd</groupId>
            <artifactId>serenity-core</artifactId>
            <version>${serenity.version}</version>
        </dependency>
        <dependency>
            <groupId>net.serenity-bdd</groupId>
            <artifactId>serenity-junit</artifactId>
            <version>${serenity.version}</version>
        </dependency>
        <dependency>
            <groupId>net.serenity-bdd</groupId>
            <artifactId>serenity-cucumber</artifactId>
            <version>${serenity.cucumber.version}</version>
        </dependency>
        <dependency>
            <groupId>org.slf4j</groupId>
            <artifactId>slf4j-simple</artifactId>
            <version>1.7.7</version>
        </dependency>
        <dependency>
            <groupId>junit</groupId>
            <artifactId>junit</artifactId>
            <version>4.12</version>
            <scope>test</scope>
        </dependency>
        <dependency>
            <groupId>org.assertj</groupId>
            <artifactId>assertj-core</artifactId>
            <version>1.7.0</version>
        </dependency>
    </dependencies>

    <build>
        <plugins>
            <plugin>
                <groupId>org.apache.maven.plugins</groupId>
                <artifactId>maven-surefire-plugin</artifactId>
                <version>2.20.1</version>
                <configuration>
                    <skip>false</skip>
                </configuration>
            </plugin>
            <plugin>
                <groupId>org.apache.maven.plugins</groupId>
                <artifactId>maven-compiler-plugin</artifactId>
                <version>3.2</version>
                <configuration>
                    <source>1.8</source>
                    <target>1.8</target>
                </configuration>
            </plugin>
            <plugin>
                <groupId>net.serenity-bdd.maven.plugins</groupId>
                <artifactId>serenity-maven-plugin</artifactId>
                <version>${serenity.maven.version}</version>
                <executions>
                    <execution>
                        <id>serenity-reports</id>
                        <phase>post-integration-test</phase>
                        <goals>
                            <goal>aggregate</goal>
                        </goals>
                    </execution>
                </executions>
            </plugin>
        </plugins>
    </build>



    <profiles>
        <profile>
            <id>android-profile</id>
            <activation>
                <property>
                    <name>env</name>
                    <value>android</value>
                </property>
            </activation>
            <properties>
                <appium.automationName>uiautomator2</appium.automationName>
                <appium.platformName>Android</appium.platformName>
                <appium.platformVersion>6.0</appium.platformVersion>
                <appium.deviceName>Android Emulator</appium.deviceName>
                <appium.app>/Users/vikramanna/Documents/vikram/serenity-core/serenity-appium/src/test/java/integration/resources/WordPress-debug.apk</appium.app>
                <appium.appPackage>org.wordpress.android</appium.appPackage>
                <appium.appActivity>org.wordpress.android.ui.accounts.LoginActivity</appium.appActivity>
            </properties>
        </profile>

        <profile>
            <id>ios-profile</id>
            <activation>
                <property>
                    <name>env</name>
                    <value>ios</value>
                </property>
            </activation>
            <properties>
                <appium.automationName>XCUITest</appium.automationName>
                <appium.platformName>IOS</appium.platformName>
                <appium.platformVersion>11.4</appium.platformVersion>
                <appium.deviceName>iPhone SE</appium.deviceName>
                <appium.app>/Users/vikramanna/Documents/vikram/serenity-core/serenity-appium/src/test/java/integration/resources/WordPress.app</appium.app>
            </properties>
        </profile>
    </profiles>

</project>
```

and commented out caps properties mentioned in serenity.properties file.

When I try to run script as below

`mvn clean verify -Denv=android` or  `mvn clean verify -Denv=ios`

it's throwing error

```
[main] ERROR net.thucydides.core.webdriver.WebDriverFacade - FAILED TO CREATE NEW WEBDRIVER_DRIVER INSTANCE class io.appium.java_client.AppiumDriver: Could not instantiate new WebDriver instance of type class io.appium.java_client.AppiumDriver (The appium.platformName needs to be specified (either IOS or ANDROID)
net.thucydides.core.webdriver.UnsupportedDriverException: Could not instantiate new WebDriver instance of type class io.appium.java_client.AppiumDriver (The appium.platformName needs to be specified (either IOS or ANDROID)
```

Am I doing something wrong wrt profiles section in pom.xml ?

**Comment by vikramvi** [comment]
@wakaleo can you please have a quick look, I'm struck from last 2 days without any progress.  Googling didn't help either to find mystery of values not being populated correctly in serenity.properties

**Comment by wakaleo** [comment]
You could do this with environment-specify property sets in the serenity.conf file.

**Comment by Joekannan** [comment]
@vikramvi - may i know you found solution for the above problem.  If so could you please share it here

**Comment by vuthelinh** [comment]
Base on your question. I think it not difficult. You can creat 2 properties files: `android.properties` and `ios.properties`

For common configuration you can put it into `serenity.conf`

To switch context. Just add more paramater in the mvn command.

`mvn verify -Dproperties=android.properties`

You can creat profile in the pom.xml.

Example u have 2 profiles with id is android or ios. To use it. You can pass parameter like this

`mvn verify -Pandroid` or `mvn verify -Pios`

**Comment by Joekannan** [comment]
@vuthelinh - Thank you for the input. I have tried this option. Created 2 property files. one for iOS and another for Android. when I run "mvn clean verify -Dproperties=Android.properties" or "mvn clean verify -Dproperties=iOS.properties" it is execution the feature files mentioned in the two runner classes. But I am expecting this command should run only one test runner file either iOS or Android as per the mvn command.

Note: I have two test runner class file. One for iOS and another for Android.

I don have any clue why this is happening. Could you please assist me on this. 

Adding my pom.xml here for reference.

<?xml version="1.0" encoding="UTF-8"?>
<project xmlns="http://maven.apache.org/POM/4.0.0" xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
         xsi:schemaLocation="http://maven.apache.org/POM/4.0.0 http://maven.apache.org/xsd/maven-4.0.0.xsd">
    <modelVersion>4.0.0</modelVersion>
    <groupId>com.TWG.Mobile</groupId>
    <artifactId>serenityFrameWork</artifactId>
    <version>1.0-SNAPSHOT</version>
    <packaging>jar</packaging>

    <name>Serenity project using Cucumber and WebDriver</name>

    <properties>
        <project.build.sourceEncoding>UTF-8</project.build.sourceEncoding>
        <serenity.version>2.0.40</serenity.version>
        <serenity.cucumber.version>1.9.38</serenity.cucumber.version>
        <project.reporting.outputEncoding>UTF-8</project.reporting.outputEncoding>
        <webdriver.driver>appium</webdriver.driver>
    </properties>

    <repositories>
      <repository>
        <snapshots>
        <enabled>false</enabled>
        </snapshots>
        <id>central</id>
        <name>bintray</name>
        <url>http://jcenter.bintray.com</url>
      </repository>
    </repositories>
    <pluginRepositories>
      <pluginRepository>
        <snapshots>
        <enabled>false</enabled>
        </snapshots>
        <id>central</id>
        <name>bintray-plugins</name>
        <url>http://jcenter.bintray.com</url>
      </pluginRepository>
    </pluginRepositories>

    <dependencies>
        <dependency>
            <groupId>net.serenity-bdd</groupId>
            <artifactId>serenity-core</artifactId>
            <version>${serenity.version}</version>
            <scope>compile</scope>
        </dependency>
        <dependency>
            <groupId>net.serenity-bdd</groupId>
            <artifactId>serenity-cucumber</artifactId>
            <version>${serenity.cucumber.version}</version>
            <scope>test</scope>
        </dependency>
        <dependency>
            <groupId>net.serenity-bdd</groupId>
            <artifactId>serenity-junit</artifactId>
            <version>${serenity.version}</version>
            <scope>test</scope>
        </dependency>
        <dependency>
            <groupId>org.slf4j</groupId>
            <artifactId>slf4j-log4j12</artifactId>
            <version>1.7.7</version>
            <scope>test</scope>
        </dependency>
        <dependency>
            <groupId>junit</groupId>
            <artifactId>junit</artifactId>
            <version>4.12</version>
            <scope>test</scope>
        </dependency>
        <dependency>
            <groupId>org.assertj</groupId>
            <artifactId>assertj-core</artifactId>
            <version>1.7.0</version>
            <scope>test</scope>
        </dependency>      
    </dependencies>
    
	<profiles>
	    <profile>
	        <id>Android</id>
	        <build>
			    <plugins>
			    	<plugin>
				      	<groupId>org.apache.maven.plugins</groupId>
				        <artifactId>maven-failsafe-plugin</artifactId>
		                <version>2.19.1</version>
		                <configuration>
		                    <includes>
		                    	<include>com/TWG/Mobile/testRunners/TestRunners.java</include>
		                    </includes>
		                    <systemPropertyVariables>
                        		<webdriver.driver>${webdriver.driver}</webdriver.driver>
                    		</systemPropertyVariables>
		                    <argLine>-Xmx512m</argLine>
		                </configuration>
		                <executions>
		                    <execution>
		                        <goals>
		                            <goal>integration-test</goal>
		                            <goal>verify</goal>
		                        </goals>
		                    </execution>
		                </executions>
			     	</plugin>
    			</plugins>
  			</build>
	    </profile>
	    <profile>
	        <id>iOS</id>
	        <build>
			    <plugins>
			    	<plugin>
				      	<groupId>org.apache.maven.plugins</groupId>
				        <artifactId>maven-failsafe-plugin</artifactId>
		                <version>2.19.1</version>
		                <configuration>
		                    <includes>
		                    	<include>com/TWG/Mobile/testRunners/TestRunnersiOS.java</include>
		                    </includes>
		                    <systemPropertyVariables>
                        		<webdriver.driver>${webdriver.driver}</webdriver.driver>
                    		</systemPropertyVariables>
		                    <argLine>-Xmx512m</argLine>
		                </configuration>
		                <executions>
		                    <execution>
		                        <goals>
		                            <goal>integration-test</goal>
		                            <goal>verify</goal>
		                        </goals>
		                    </execution>
		                </executions>
			     	</plugin>
    			</plugins>
  			</build>
	    </profile>
	</profiles>
	
    <build>
        <plugins>
            <plugin>
                <groupId>com.github.temyers</groupId>
                <artifactId>cucumber-jvm-parallel-plugin</artifactId>
                <version>4.2.0</version>
                <executions>
                    <execution>
                        <id>generateRunners</id>
                        <phase>generate-test-sources</phase>
                        <goals>
                            <goal>generateRunners</goal>
                        </goals>
                        <configuration>
                            <glue>
                                <package>com.TWG.Mobile.Cucumber.StepDef</package>
                            </glue>
                            <parallelScheme>FEATURE</parallelScheme>
                            <customVmTemplate>src/test/resources/cucumber-serenity-runner.vm</customVmTemplate>
                        </configuration>
                    </execution>
                </executions>
            </plugin>
            <plugin>
                <groupId>org.apache.maven.plugins</groupId>
                <artifactId>maven-compiler-plugin</artifactId>
                <version>3.2</version>
                <configuration>
                    <source>1.8</source>
                    <target>1.8</target>
                </configuration>
            </plugin>
            <plugin>
                <groupId>net.serenity-bdd.maven.plugins</groupId>
                <artifactId>serenity-maven-plugin</artifactId>
                <version>${serenity.version}</version>
                <dependencies>
                     <dependency>
                        <groupId>net.serenity-bdd</groupId>
                        <artifactId>serenity-core</artifactId>
                        <version>${serenity.version}</version>
                    </dependency>
                </dependencies>
                <executions>
                    <execution>
                        <id>serenity-reports</id>
                        <phase>post-integration-test</phase>
                        <goals>
                            <goal>aggregate</goal>
                        </goals>
                    </execution>
                </executions>
            </plugin>
        </plugins>
        <pluginManagement>
        	<plugins>

        		<plugin>
        			<groupId>org.eclipse.m2e</groupId>
        			<artifactId>lifecycle-mapping</artifactId>
        			<version>1.0.0</version>
        			<configuration>
        				<lifecycleMappingMetadata>
        					<pluginExecutions>
        						<pluginExecution>
        							<pluginExecutionFilter>
        								<groupId>
        									com.github.temyers
        								</groupId>
        								<artifactId>
        									cucumber-jvm-parallel-plugin
        								</artifactId>
        								<versionRange>
        									[4.2.0,)
        								</versionRange>
        								<goals>
        									<goal>generateRunners</goal>
        								</goals>
        							</pluginExecutionFilter>
        							<action>
        								<ignore></ignore>
        							</action>
        						</pluginExecution>
        					</pluginExecutions>
        				</lifecycleMappingMetadata>
        			</configuration>
        		</plugin>
        	</plugins>
        </pluginManagement>
    </build>
</project>

**Comment by vuthelinh** [comment]
I suggest your pom.xml is correct. 
For running on Android, pls try:

`mvn clean verify -Dproperties=Android.properties -PAndroid`

**Comment by vuthelinh** [comment]
The reason the two runner classes is running at the same time. I think because `cucumber-jvm-parallel-plugin`

So if you run `mvn clean verify ....` it will use build phase outside of each profile.

**Comment by Joekannan** [comment]
Hello @vuthelinh - I commented the "cucumber-jvm-parallel-plugin" in pom.xml, still the feature file mentioned in two runner class is getting executed.

I am using "mvn clean verify -Dproperties=Android.properties -PAndroid" still It executes iOS runner file and getting failed :(. Any other way to fix it.

**Comment by Joekannan** [comment]
@vuthelinh - Though I add excludes tag in profile to exclude one profile from other, it is not working. 

<profiles>
	    <profile>
	        <id>Android</id>
	        <activation>
      			<activeByDefault>true</activeByDefault>
    		</activation>
	        <build>
			    <plugins>
			    	<plugin>
				      	<groupId>org.apache.maven.plugins</groupId>
				        <artifactId>maven-failsafe-plugin</artifactId>
		                <version>2.19.1</version>
		                <executions>
		                    <execution>
			                    <configuration>
				                    <includes>
				                    	<include>com/TWG/Mobile/testRunners/TestRunners.java</include>
				                    </includes>
				                    <excludes>
				                    	<include>com/TWG/Mobile/testRunners/TestRunnersiOS.java</include>
				                    </excludes>
				                    <systemPropertyVariables>
		                        		<webdriver.driver>${webdriver.driver}</webdriver.driver>
		                    		</systemPropertyVariables>
				                    <argLine>-Xmx512m</argLine>
			                	</configuration>
		                        <goals>
		                            <goal>integration-test</goal>
		                            <goal>verify</goal>
		                        </goals>
		                    </execution>
		                </executions>
			     	</plugin>
    			</plugins>
  			</build>
	    </profile>
	    <profile>
	        <id>iOS</id>
	        <activation>
      			<activeByDefault>true</activeByDefault>
    		</activation>
	        <build>
			    <plugins>
			    	<plugin>
				      	<groupId>org.apache.maven.plugins</groupId>
				        <artifactId>maven-failsafe-plugin</artifactId>
		                <version>2.19.1</version>
		                <executions>
		                    <execution>
			                    <configuration>
				                    <includes>
				                    	<include>com/TWG/Mobile/testRunners/TestRunnersiOS.java</include>
				                    </includes>
				                    <excludes>
				                    	<include>com/TWG/Mobile/testRunners/TestRunners.java</include>
				                    </excludes>
				                    <systemPropertyVariables>
		                        		<webdriver.driver>${webdriver.driver}</webdriver.driver>
		                    		</systemPropertyVariables>
				                    <argLine>-Xmx512m</argLine>
			                	</configuration>
		                        <goals>
		                            <goal>integration-test</goal>
		                            <goal>verify</goal>
		                        </goals>
		                    </execution>
		                </executions>
			     	</plugin>
    			</plugins>
  			</build>
	    </profile>
	</profiles>

**Comment by Joekannan** [comment]
@vuthelinh - Though I add excludes tag in my profiles to exclude one runner from other profile, test executed for both the profiles.

<profiles>
	    <profile>
	        <id>Android</id>
	        <activation>
      			<activeByDefault>true</activeByDefault>
    		</activation>
	        <build>
			    <plugins>
			    	<plugin>
				      	<groupId>org.apache.maven.plugins</groupId>
				        <artifactId>maven-failsafe-plugin</artifactId>
		                <version>2.19.1</version>
		                <executions>
		                    <execution>
			                    <configuration>
				                    <includes>
				                    	<include>com/TWG/Mobile/testRunners/TestRunners.java</include>
				                    </includes>
				                    <excludes>
				                    	<include>com/TWG/Mobile/testRunners/TestRunnersiOS.java</include>
				                    </excludes>
				                    <systemPropertyVariables>
		                        		<webdriver.driver>${webdriver.driver}</webdriver.driver>
		                    		</systemPropertyVariables>
				                    <argLine>-Xmx512m</argLine>
			                	</configuration>
		                        <goals>
		                            <goal>integration-test</goal>
		                            <goal>verify</goal>
		                        </goals>
		                    </execution>
		                </executions>
			     	</plugin>
    			</plugins>
  			</build>
	    </profile>
	    <profile>
	        <id>iOS</id>
	        <activation>
      			<activeByDefault>true</activeByDefault>
    		</activation>
	        <build>
			    <plugins>
			    	<plugin>
				      	<groupId>org.apache.maven.plugins</groupId>
				        <artifactId>maven-failsafe-plugin</artifactId>
		                <version>2.19.1</version>
		                <executions>
		                    <execution>
			                    <configuration>
				                    <includes>
				                    	<include>com/TWG/Mobile/testRunners/TestRunnersiOS.java</include>
				                    </includes>
				                    <excludes>
				                    	<include>com/TWG/Mobile/testRunners/TestRunners.java</include>
				                    </excludes>
				                    <systemPropertyVariables>
		                        		<webdriver.driver>${webdriver.driver}</webdriver.driver>
		                    		</systemPropertyVariables>
				                    <argLine>-Xmx512m</argLine>
			                	</configuration>
		                        <goals>
		                            <goal>integration-test</goal>
		                            <goal>verify</goal>
		                        </goals>
		                    </execution>
		                </executions>
			     	</plugin>
    			</plugins>
  			</build>
	    </profile>
	</profiles>

**Comment by Joekannan** [comment]
<profiles>
	    <profile>
	        <id>Android</id>
	        <activation>
      			<activeByDefault>true</activeByDefault>
    		</activation>
	        <build>
			    <plugins>
			    	<plugin>
				      	<groupId>org.apache.maven.plugins</groupId>
				        <artifactId>maven-failsafe-plugin</artifactId>
		                <version>2.19.1</version>
		                <executions>
		                    <execution>
			                    <configuration>
				                    <includes>
				                    	<include>com/TWG/Mobile/testRunners/TestRunners.java</include>
				                    </includes>
				                    <excludes>
				                    	<include>com/TWG/Mobile/testRunners/TestRunnersiOS.java</include>
				                    </excludes>
				                    <systemPropertyVariables>
		                        		<webdriver.driver>${webdriver.driver}</webdriver.driver>
		                    		</systemPropertyVariables>
				                    <argLine>-Xmx512m</argLine>
			                	</configuration>
		                        <goals>
		                            <goal>integration-test</goal>
		                            <goal>verify</goal>
		                        </goals>
		                    </execution>
		                </executions>
			     	</plugin>
    			</plugins>
  			</build>
	    </profile>
	    <profile>
	        <id>iOS</id>
	        <activation>
      			<activeByDefault>true</activeByDefault>
    		</activation>
	        <build>
			    <plugins>
			    	<plugin>
				      	<groupId>org.apache.maven.plugins</groupId>
				        <artifactId>maven-failsafe-plugin</artifactId>
		                <version>2.19.1</version>
		                <executions>
		                    <execution>
			                    <configuration>
				                    <includes>
				                    	<include>com/TWG/Mobile/testRunners/TestRunnersiOS.java</include>
				                    </includes>
				                    <excludes>
				                    	<include>com/TWG/Mobile/testRunners/TestRunners.java</include>
				                    </excludes>
				                    <systemPropertyVariables>
		                        		<webdriver.driver>${webdriver.driver}</webdriver.driver>
		                    		</systemPropertyVariables>
				                    <argLine>-Xmx512m</argLine>
			                	</configuration>
		                        <goals>
		                            <goal>integration-test</goal>
		                            <goal>verify</goal>
		                        </goals>
		                    </execution>
		                </executions>
			     	</plugin>
    			</plugins>
  			</build>
	    </profile>
	</profiles>

**Comment by vuthelinh** [comment]
@Joekannan Can you share your demo project ?

**Comment by Joekannan** [comment]
@Jacobvu84 - This is my repo. I have uploaded my demo project here.
https://github.com/Joekannan/SerenityDemoProject. Thank you for your help in advance.

**Comment by Jacobvu84** [comment]
@Joekannan Sorry, No time to review deeply.  It seems to be difficult for me.

By the way, this is my project demo for [mobile](https://github.com/Jacobvu84/serenity-core/tree/master/serenity-mobile-screenplay-example)

To run on android: `mvn clean verify -Dproperties=android.properties -Pcucumber`

Base on this sample, pls try with your app. If possible, give me your apk and ipa, I will create sample project for you.

**Comment by emrebaygul** [comment]
How can I run in serenity with multiple android devices?

**Comment by Jacobvu84** [comment]
@emrebaygul 

If you are using screenplay with Junit

#### sereinty.properties 
or any name you want such as android.properties

You need to add more information below
```
....
webdriver.driver=appium
manage.appium.servers = true
appium.deviceNames = udid-device1,udid-device2
.....
```

#### pom.xml
you should add more this configuration for `maven-failsafe-plugin`
```
<argLine>Xmx1024m</argLine>
<parallel>classes</parallel>
<threadCount>2</threadCount>
```

That is all configuration you need to run your test in parallel. 

Note: Sometimes we should to face any issue such as version conflict, libraries compatibility ( java-client vs appium server).... so forth. Pls try googling

**Comment by ITester2020** [comment]
Hello, 
I'm using Serinity +Appium for android emulator testing. I have set serenity property file but where i can write the code to invoke the emulator.
